lib/integration.go

```diff
@@ -0,0 +1,136 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type IIntegration interface {
+	Create(ctx context.Context, request CreateIntegrationRequest) (*IntegrationResponse, error)
+	GetAll(ctx context.Context) (*GetIntegrationsResponse, error)
+	GetActive(ctx context.Context) (*GetIntegrationsResponse, error)
+	Update(ctx context.Context, integrationId string, request UpdateIntegrationRequest) (*IntegrationResponse, error)
+	Delete(ctx context.Context, integrationId string) (*IntegrationResponse, error)
+}
+
+type IntegrationService service
+
+func (integration *IntegrationService) Create(ctx context.Context, request CreateIntegrationRequest) (*IntegrationResponse, error) {
```

Comment:
Update and abbreviate the receiver, use `i` instead of `integration`, and also remove the reference pointer.

Reply:
Should I  remove the pointer reference return type for all methods?

Reply:
@akhill10 nope, not the return type, I mean this
`func (integration *IntegrationService) Create(ctx context.Context, request CreateIntegrationRequest) (*IntegrationResponse, error) {}`

to 

`func (i IntegrationService) Create(ctx context.Context, request CreateIntegrationRequest) (*IntegrationResponse, error) {}`